sodium/src/secbuf.rs

```diff
@@ -232,16 +232,48 @@ impl SecBuf {
     /// helper for writing data to our internal buffer
     pub fn write(&mut self, offset: usize, data: &[u8]) -> Result<(), SodiumError> {
         if offset + data.len() > self.len() {
-            return Err(SodiumError::new("bad write offset / length"));
+            return Err(SodiumError::new("SecBuf write overflow"));
         }
-
         unsafe {
             let mut b = self.write_lock();
             std::ptr::copy(data.as_ptr(), (**b).as_mut_ptr().add(offset), data.len());
         }
+        Ok(())
+    }
+
+    // -- Helpers -- //
+
+    /// Check if the buffer is empty i.e. [0,0,0,0,0,0,0,0]
+    pub fn is_empty(&mut self) -> bool {
+        let buf = self.read_lock();
+        for i in 0..buf.len() {
+            if buf[i] != 0 {
+                return false;
+            }
+        }
+        true
+    }
 
+    /// Load the [u8] into the SecBuf
+    pub fn from_array(&mut self, data: &[u8]) -> Result<(), SodiumError> {
+        if (data.len() != self.len()) {
+            return Err(SodiumError::Generic(
+                "Input does not have same size as SecBuf".to_string(),
+            ));
+        }
+        let mut buf = self.write_lock();
+        for x in 0..data.len() {
+            buf[x] = data[x];
```

Comment:
should not do this with iteration... please see the `write` function on secbuf. In fact, I don't think we need this function... just use `buf.write(0, &[ /* ... */])`